src/nu/validator/saxtree/DocumentFragment.java

```diff
@@ -37,7 +37,7 @@
      * The constructor.
      */
     public DocumentFragment() {
-        super(new LocatorImpl());
+        super(new Locator2Impl());
```

Comment:
`Locator2Impl` is not a drop-in replacement for `LocatorImpl`. At least, `getXMLVersion()` is going to return `null` with the former and `1.0` with the latter.